### PR TITLE
feat(add): disable emoji flag

### LINKED ISSUE #304: flag to enable or disable emojis
See #294 for context

### PR DESCRIPTION
**Pull request by sendilkumarn**
fix #304 

r? @radu-matei

### CODE REVIEW

File: pkg/cmdline/cmdline.go [hunk 1]
```diff
@@ -156,14 +157,21 @@ func progress(cli *cmdline, app, desc string, codes <-chan builder.SummaryStatus
...
+	return fmt.Sprintf("%s: %s", red(msg), concatStrAndEmoji("FAIL", " ❌ ", disableEmoji))
+}
+
+func concatStrAndEmoji(text string, emoji string, disableEmoji bool) string {
+	var concatStr bytes.Buffer
+	concatStr.WriteString(text)
+	if !disableEmoji {
+		concatStr.WriteString(emoji)
+	}
+	concatStr.String()
```
Comment (radu-matei) [inline pkg/cmdline/cmdline.go]: This should probably return.

File: pkg/cmdline/cmdline.go [hunk 2]
```diff
@@ -156,14 +157,21 @@ func progress(cli *cmdline, app, desc string, codes <-chan builder.SummaryStatus
...
 
-func failStr(msg string) string {
-	const fail = "FAIL " + "❌"
-	return fmt.Sprintf("%s: %s", red(msg), fail)
+func failStr(msg string, disableEmoji bool) string {
+	return fmt.Sprintf("%s: %s", red(msg), concatStrAndEmoji("FAIL", " ❌ ", disableEmoji))
+}
+
+func concatStrAndEmoji(text string, emoji string, disableEmoji bool) string {
+	var concatStr bytes.Buffer
```
Comment (bacongobbler) [inline pkg/cmdline/cmdline.go]: more of an FYI, but you can use a [`strings.Builder`](https://golang.org/pkg/strings/#Builder) here instead for more efficient memory usage.

File: pkg/cmdline/cmdline.go [hunk 3]
```diff
@@ -156,14 +157,21 @@ func progress(cli *cmdline, app, desc string, codes <-chan builder.SummaryStatus
...
+	concatStr.WriteString(text)
+	if !disableEmoji {
```
Comment (itowlson) [inline pkg/cmdline/cmdline.go]: I find the use of `disableEmoji` throughout a bit hard to follow, and this double negative is I think why - there is an extra cognitive load of 'and pass whether we're _not_ showing emoji' and 'if we're **not** not showing emoji...".  Consider a positive name `displayEmoji` and a positive condition check.

File: pkg/cmdline/cmdline.go (hunk 2 above)
Comment (sendilkumarn) [reply pkg/cmdline/cmdline.go]: Thanks, it was introduced in 1.10 right? Gotcha

File: pkg/cmdline/cmdline.go (hunk 3 above)
Comment (sendilkumarn) [reply pkg/cmdline/cmdline.go]: Agree to it 👍  changed it

**Comment by radu-matei** [comment]
Now that #874 is merged, a rebase from `master` should make the CI go green 😄

**Comment by sendilkumarn** [comment]
@radu-matei Done 👍

**Comment by radu-matei** [comment]
Manually tested again, LGTM.

**Comment by radu-matei** [comment]
Thanks for your contribution, @sendilkumarn!
Let us know when you want to pick up something else here, we'd be more than happy to help 😄
